File: Euler 4/Euler4.java

public class Euler4 {
	public static void main(String[] args) {
		System.out.println(largestPalindrome());
	}
	
	private static int largestPalindrome() {
		int max = 0;
		for (int i = 999; i >= 100; i--) {
			for (int j = 999; j >= 100; j--) {
				int num = i * j;
				if (parse(num)) {
					if (num > max) {
						max = num;
					}
				}
			}
		}
		return max;
	}
	
	private static Boolean parse(Integer num) {
		String str = Integer.toString(num);
		for (int k = 0; k < str.length()/2; k++) {
			if (str.charAt(k) != str.charAt(str.length() - 1 - k)) return false;
		}
		return true;
	}
}

File: Euler 4/Euler4.js
/**
 * Created by eddolan on 1/3/15.
 */
var isPalendrom = function(num){
    var numArray = num.toString().split('');
    while (numArray.length > 1){
        if (numArray.pop() != numArray.shift()){
            return false
        }
    }
    return true;
};

var euler4Iter = function() {
    var max = 0;
    for (var x = 999; x >= 100; x--) {
        for (var y = 999; y >= 100; y--) {
            testNum = x * y;
            if (isPalendrom(testNum) && testNum > max) {
                max = testNum;
            }
        }
    }
    return max;
};
console.log(euler4Iter());